### PR TITLE
Mvp/authentication mentor experience

### PR DESCRIPTION
**Pull request by OUMontiel**
# What does this PR do?
- Adds multiple select to form in signup page for mentors to select their experience and stores that selection in database

**Related Issues**
- #30 

---
**Merge Checklist**
- [x] Branch is up to date
- [ ] No code style issues

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/MentorSignupServlet.java [hunk 1]
```diff
@@ -74,6 +76,27 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
+      PreparedStatement preparedStatement = connection.prepareStatement(query);
+      preparedStatement.setInt(1, Utility.getUserId());
+      preparedStatement.setInt(2, Integer.parseInt(tag));
+      preparedStatement.execute();
+      connection.close();
+    } catch (SQLException exception) {
+      // If the connection or the query don't go through, get the log of the error.
+      Logger logger = Logger.getLogger(MentorSignupServlet.class.getName());
+      logger.log(Level.SEVERE, exception.getMessage(), exception);
+    }
```
Comment (shaargtz) [inline src/main/java/com/google/sps/servlets/MentorSignupServlet.java:97]: indent two spaces to the right

Comment (OUMontiel) [reply src/main/java/com/google/sps/servlets/MentorSignupServlet.java]: Done!

File: src/main/java/com/google/sps/classes/Utility.java [hunk 2]
```diff
@@ -52,7 +52,7 @@ public static int getUserId() {
     String email = userService.getCurrentUser().getEmail();
 
     // If user is not logged in, return -1.
-    if (email = "") {
+    if (email == "") {
```
Comment (timothy-parsons) [inline src/main/java/com/google/sps/classes/Utility.java:55]: I believe you should always use .equals for string comparison in Java - https://stackoverflow.com/questions/513832/how-do-i-compare-strings-in-java

File: src/main/java/com/google/sps/servlets/MenteeSignupServlet.java [hunk 3]
```diff
@@ -49,7 +49,7 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     String password = "";
 
     // Set up query to insert new user into database.
-    String query = "INSERT INTO User (fname, lname, username, email, major_id, is_mentor) "
+    String query = "INSERT INTO User (first_name, last_name, username, email, major_id, is_mentor) "
         + "VALUES (?, ?, ?, ?, ?, ?)";
 
     try {
```
Comment (timothy-parsons) [inline src/main/java/com/google/sps/servlets/MenteeSignupServlet.java:55]: This is getting to be a very long method. Consider ways  you can break this up into helper methods (my first intuition is the stuff inside the the try and catch blocks)

File: src/main/java/com/google/sps/servlets/MentorSignupServlet.java [hunk 4]
```diff
@@ -74,6 +76,27 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
+      try {
+      // Establish connection to MySQL database.
+      Connection connection = DriverManager.getConnection(url, user, password);
+
+      // Create the MySQL INSERT prepared statement.
+      PreparedStatement preparedStatement = connection.prepareStatement(query);
+      preparedStatement.setInt(1, Utility.getUserId());
+      preparedStatement.setInt(2, Integer.parseInt(tag));
+      preparedStatement.execute();
+      connection.close();
```
Comment (timothy-parsons) [inline src/main/java/com/google/sps/servlets/MentorSignupServlet.java:92]: indent two spaces

File: src/main/java/com/google/sps/servlets/MentorSignupServlet.java [hunk 5]
```diff
@@ -41,6 +42,7 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     String username = request.getParameter("username");
```
Comment (timothy-parsons) [inline src/main/java/com/google/sps/servlets/MentorSignupServlet.java:42]: Same with this method - it's very long. Break this up into multiple helper functions - it'll be helpful for readability, and it'll be easier for all of you to figure out where things are actually happening in your code.

File: src/mysql/create.sql [hunk 6]
```diff
@@ -27,13 +27,20 @@ CREATE TABLE User (
...
-  user_id INT NOT NULL,
   message TEXT,
   url VARCHAR(255),
   date_time DATETIME,
-  PRIMARY KEY (id),
-  FOREIGN KEY (user_id) 
-  REFERENCES User (id)
+  PRIMARY KEY (id)
+);
+
```
Comment (timothy-parsons) [inline src/mysql/create.sql:35]: Consider adding comments that describe the purpose of each of these tables - can be short and sweet.

File: src/mysql/populate.sql [hunk 7]
```diff
@@ -13,7 +13,7 @@ VALUES
   ('ITI'),
```
Comment (timothy-parsons) [inline src/mysql/populate.sql:13]: nit - if this file is populating dummy data, that should be clear by the filename. Either it should be included in a testdata directory of the filename should reference that it is populating dummy data.

File: tests/java-db-test/mysql/create.sql [hunk 8]
```diff
@@ -25,13 +25,20 @@ CREATE TABLE User (
...
-  user_id INT NOT NULL,
   message TEXT,
   url VARCHAR(255),
   date_time DATETIME,
-  PRIMARY KEY (id),
-  FOREIGN KEY (user_id) 
-  REFERENCES User (id)
+  PRIMARY KEY (id)
+);
+
```
Comment (timothy-parsons) [inline tests/java-db-test/mysql/create.sql:33]: Same as the file above - consider adding comments that describe these tables.

File: src/main/java/com/google/sps/classes/Utility.java (hunk 2 above)
Comment (OUMontiel) [reply src/main/java/com/google/sps/classes/Utility.java]: That's right! We just had an issue with that and I completely forgot! Fixed it here but will also make a hotfix about it, might be causing issues to the others.

File: src/main/java/com/google/sps/servlets/MenteeSignupServlet.java (hunk 3 above)
Comment (OUMontiel) [reply src/main/java/com/google/sps/servlets/MenteeSignupServlet.java]: This method was the same as part of the method in MentorSignupServlet.java, so I created a Utility method they both use.

File: src/main/java/com/google/sps/servlets/MentorSignupServlet.java (hunk 4 above)
Comment (OUMontiel) [reply src/main/java/com/google/sps/servlets/MentorSignupServlet.java:68]: Done!

File: src/main/java/com/google/sps/servlets/MentorSignupServlet.java (hunk 5 above)
Comment (OUMontiel) [reply src/main/java/com/google/sps/servlets/MentorSignupServlet.java:45]: Done!

File: src/mysql/create.sql (hunk 6 above)
Comment (OUMontiel) [reply src/mysql/create.sql]: Ok, I will! But I'll make all the SQL changes in another PR, I just had to make small changes to the database for my feature to work.

File: src/mysql/populate.sql (hunk 7 above)
Comment (OUMontiel) [reply src/mysql/populate.sql:13]: Same, I'll make SQL changes in another PR.

File: tests/java-db-test/mysql/create.sql (hunk 8 above)
Comment (OUMontiel) [reply tests/java-db-test/mysql/create.sql]: > Same, I'll make SQL changes in another PR.

**Comment by OUMontiel** [comment]
Noted on the size of the PR, will work on getting them smaller! I did not apply any of the SQL changes because they apply to the issue this PR is linked too, but I'll work on them separately.

File: src/mysql/populate.sql (hunk 7 above)
Comment (timothy-parsons) [reply src/mysql/populate.sql:13]: If you're going to wait on those SQL changes can you add TODOs?

Comment (OUMontiel) [reply src/mysql/populate.sql:13]: Sure!

**Pull request merged**